kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java

```diff
@@ -343,6 +345,254 @@ public void testGetAndUpdateContainer() {
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerOneToDeploy);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerTwoToDeploy);
+
+        containerOneToDeploy.setReleasedId(releaseId101);
+
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), TWO_ID, containerOneToDeploy);
+            fail("Container one was updated from container two REST endpoint.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(400, e.getResponseCode());
+            KieServerAssert.assertResultContainsString(e.getMessage(), "can't updated container with id " + ONE_ID + " on container " + TWO_ID);
+        }
+
+        // Check container that are not changed
+        ContainerSpec containerResponseEntity = mgmtControllerClient.getContainerInfo(serverTemplate.getId(), ONE_ID);
+        assertEquals(ONE_ID, containerResponseEntity.getId());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+        assertEquals(KieContainerStatus.STOPPED, containerResponseEntity.getStatus());
+
+        containerResponseEntity = mgmtControllerClient.getContainerInfo(serverTemplate.getId(), TWO_ID);
+        assertEquals(TWO_ID, containerResponseEntity.getId());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+        assertEquals(KieContainerStatus.STOPPED, containerResponseEntity.getStatus());
+    }
+
+    @Test
+    public void testUpdateContainerServerTemplateId() {
+        // Create kie server instance connection in controller.
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        // Deploy container for kie server instance.
+        ContainerSpec containerToDeploy = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerToDeploy);
+
+        // Get container using kie controller.
+        ContainerSpec containerResponseEntity = mgmtControllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        checkContainer(containerResponseEntity, KieContainerStatus.STOPPED);
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+
+        // Create other server template.
+        ServerTemplate secondTemplate = new ServerTemplate("st-id", "st-id");
+        mgmtControllerClient.saveServerTemplate(secondTemplate);
+
+        // Update container
+        containerToDeploy.setServerTemplateKey(secondTemplate);
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), containerToDeploy);
+            fail("Should throw exception that is not posible to change server template key");
```

Comment:
This can be reproducer for [JBPM-5930](https://issues.jboss.org/browse/JBPM-5930)